src/Data/Factory.php

```diff
@@ -0,0 +1,25 @@
+<?php
+/* Copyright (c) 2017 Richard Klees <richard.klees@concepts-and-training.de> Extended GPL, see docs/LICENSE */
+
+namespace ILIAS\Data;
+
+/**
+ * Builds data types.
```

Comment:
I think an `Option` type would be a candidate to go here. And `datetime` of course. We also might have applications for some more restricted (wrt to phps `array`) types of lists or dictionaries. I also could imagine that we could put "HTML" or "Text" or "Markdown" types here to get rid of the dreaded "did we escape yet" problem in emails. We also could put types for IDs here to make it impossible to mix up obj/ref_ids. None of these would derive from `Result`.